File: backend/graph/model/experience.go
package model

type Experience struct {
	ID             string `json:"id" gorm:"primaryKey"`
	UserID         string `json:"user_id"`
	Title          string `json:"title"`
	EmploymentType string `json:"type"`
	CompanyName    string `json:"name"`
	Location       string `json:"location"`
	IsActive       string `json:"is_active"`
	StartYear      string `json:"start_date"`
	EndYear        string `json:"end_date"`
	Industry       string `json:"industry"`
	Headline       string `json:"headline"`
}

File: LinkhedIn/src/query/experience.ts
import { gql } from "@apollo/client";

export const EXPERIENCE_USER_QUERY = gql`
query UserExperience($id:String!){
  userExperience(id:$id){
    User{
      id
      name
      email
    }
    Title
    EmploymentType
    CompanyName
    Location
    IsActive
    StartYear
    EndYear
    Industry
    Headline
    ID
  }
}
`

export const CREATE_EXPERIENCE_QUERY = gql`
mutation CreateExperience($input:NewExperience!){
  createExperience(input:$input)
}
`

export const DELETE_EXPERIENCE_QUERY = gql`
  mutation DeleteExperience($id:String!){
    deleteExperience(id: $id)
}

`

export const UPDATE_EXPERIENCE_QUERY = gql`
mutation UpdateExperience($id:String!, $input:NewExperience!){
  updateExperience(id:$id, input:$input)
}
`